src/Language/PureScript/TypeChecker/Types.hs

```diff
@@ -496,6 +499,17 @@ inferLetBinding seen (BindingGroupDeclaration ds : rest) ret j = do
   bindNames dict $ do
     makeBindingGroupVisible
     inferLetBinding (seen ++ [BindingGroupDeclaration ds']) rest ret j
+inferLetBinding seen (TypeSynonymDeclaration sa@(ss, _) name args ty : rest) ret j = do
+  moduleName <- unsafeCheckCurrentModule
+  (kind', ty') <- warnAndRethrow (addHint (ErrorInTypeSynonym name) . addHint (positionedError ss)) $ do
+    checkDuplicateTypeArguments $ map fst args
+    traverse replaceAllTypeSynonyms . swap =<< kindOfLocalTypeSynonym moduleName (sa, name, args, ty)
```

Comment:
Does there need to be an `introduceSkolemScope` in here? I tried it with and without and it didn't seem to make a difference on any tests I could come up with, but it seems to be paired with all the other `replaceAllTypeSynonyms`s in this file.